Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h

```diff
@@ -798,23 +798,16 @@ insert_subconstraint(Vertex_handle vaa,
   // edges may contain mirror edges. They no longer exist after triangulate_hole
   // so we have to remove them before calling get_bounded_faces
   if(! edges.empty()){
-    typename List_edges::iterator it, it2;
-    
-    it = edges.begin();
-    it2 = it;
-    ++it2;
-    for(; it2 != edges.end();){
-      Edge e1 = *it, e2 = *it2;
-      if(this->mirror_edge(e1) == e2){
-        typename List_edges::iterator del = it;
-        --it;
-        edges.erase(del);
+
+  std::set<Face_handle> faces(intersected_faces.begin(), intersected_faces.end());
```

Comment:
`boost::flat_set`, from Boost.Container, would probably be more efficient, memory-wise. It was introduced in Boost-1.48, that is required by CGAL as the minimal version of Boost. That is safe to use it.